homeassistant/components/bond/cover.py

```diff
@@ -0,0 +1,84 @@
+"""Support for Bond covers."""
+import asyncio
+import logging
+from typing import Any, Callable, Dict, List, Optional
+
+from bond import BOND_DEVICE_TYPE_MOTORIZED_SHADES, Bond
+
+from homeassistant.components.cover import DEVICE_CLASS_SHADE, CoverEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN
+from .utils import BondDevice, get_bond_devices
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: Callable[[List[Entity]], None],
+) -> None:
+    """Set up Bond cover devices."""
+
+    bond: Bond = hass.data[DOMAIN][entry.entry_id]
+
+    async def discover():
+        devices = await get_bond_devices(hass, bond)
+        covers = [
+            BondCover(bond, device)
+            for device in devices
+            if device.type == BOND_DEVICE_TYPE_MOTORIZED_SHADES
+        ]
+        async_add_entities(covers)
+
+    asyncio.create_task(discover())
```

Comment:
im wondering if this really needs to be wrapped into a function and have a task created. you await the function which immediately awaits the get_bond_devices which seems to be only async part of this. thus it should be fine without such a function task wrapper imo

Reply:
Hi Chris. I am happy to consider alternative advise, but, IMO, creating a task is necessary. get_bond_devices() can take time, often more than 10 seconds. In that case, without wrapping task, the await is on the 'setup-platform' thread and HA will warn, or even abort the setup. Creating a task moves the get_bond_devices() call off the setup thread, and allows adding devices when API finishes without any problems.

Reply:
ah ok that makes more sense with the delays